app/assets/javascripts/wizard/lib/preview.js.es6

```diff
@@ -18,10 +18,11 @@ function canvasFor(image, w, h) {
   h = Math.ceil(h);
 
   const can = document.createElement("canvas");
-  can.width = w;
-  can.height = h;
+  can.width = w * 2;
```

Comment:
Shouldn't this be `window.devicePixelRatio` ? It would be 2 on a hi dpi display, 1 otherwise.

Reply:
(same for all the other hard coded 2s below!)

Reply:
Oh, I didn't know that API existed 😍